### PR TITLE
Migrate from moment to date-fns and update tests for eventdatepicker

### PR DESCRIPTION
**Pull request by nathanchica**
This pull request migrates the project from Moment.js to date-fns, reflecting changes across multiple files to improve performance, reduce bundle size, and leverage modern JavaScript features. Key updates include replacing Moment.js functions with date-fns equivalents, updating dependencies, and modifying related components and tests.

### Migration to date-fns:
* [`README.md`](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R18-R28): Added a migration note explaining the transition from Moment.js to date-fns, including reasons and key changes.
* [`package.json`](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L7-R7): Replaced `moment` and `@date-io/moment` dependencies with `date-fns` and `@date-io/date-fns`. [[1]](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L7-R7) [[2]](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R21-L24)

### Component updates:
* [`src/App.tsx`](diffhunk://#diff-26ad4b834941d9b19ebf9db8082bd202aaf72ea0ddea85f5a8a0cb3c729cc6f2L1-R1): Updated the MUI date adapter from `AdapterMoment` to `AdapterDateFns`. [[1]](diffhunk://#diff-26ad4b834941d9b19ebf9db8082bd202aaf72ea0ddea85f5a8a0cb3c729cc6f2L1-R1) [[2]](diffhunk://#diff-26ad4b834941d9b19ebf9db8082bd202aaf72ea0ddea85f5a8a0cb3c729cc6f2L16-R16)
* [`src/components/EventCards/EventDatepicker.tsx`](diffhunk://#diff-d3d2ad8434dd7c8308494446eb3f23d4e11ec5ceb657d8615b97645930f2aefaL11-R11): Replaced Moment.js logic with date-fns functions (`subDays`, `isSameDay`) and updated date handling to use native JavaScript `Date` objects. Added helper text for invalid date input. [[1]](diffhunk://#diff-d3d2ad8434dd7c8308494446eb3f23d4e11ec5ceb657d8615b97645930f2aefaL11-R11) [[2]](diffhunk://#diff-d3d2ad8434dd7c8308494446eb3f23d4e11ec5ceb657d8615b97645930f2aefaL22-R57) [[3]](diffhunk://#diff-d3d2ad8434dd7c8308494446eb3f23d4e11ec5ceb657d8615b97645930f2aefaR73-R74) [[4]](diffhunk://#diff-d3d2ad8434dd7c8308494446eb3f23d4e11ec5ceb657d8615b97645930f2aefaL87-R96)

### Test updates:
* [`src/components/EventCards/__tests__/CreateEventCard.test.js`](diffhunk://#diff-06b8dae2443a1b16b4de1c19aabbbd490648ba20809070d79c6cee69b699ef72L2-R2): Switched the date adapter to `AdapterDateFns`. [[1]](diffhunk://#diff-06b8dae2443a1b16b4de1c19aabbbd490648ba20809070d79c6cee69b699ef72L2-R2) [[2]](diffhunk://#diff-06b8dae2443a1b16b4de1c19aabbbd490648ba20809070d79c6cee69b699ef72L22-R22)
* [`src/components/EventCards/__tests__/EventDatepicker.test.js`](diffhunk://#diff-221a584fe73547286e0868f15a38cb85c51f67cb3527bce0b88478990f889309L1-R24): Updated tests to use date-fns, added fake timers for consistent test dates, and enhanced coverage for user interactions and edge cases. [[1]](diffhunk://#diff-221a584fe73547286e0868f15a38cb85c51f67cb3527bce0b88478990f889309L1-R24) [[2]](diffhunk://#diff-221a584fe73547286e0868f15a38cb85c51f67cb3527bce0b88478990f889309L24-R43) [[3]](diffhunk://#diff-221a584fe73547286e0868f15a38cb85c51f67cb3527bce0b88478990f889309L39-R180) [[4]](diffhunk://#diff-221a584fe73547286e0868f15a38cb85c51f67cb3527bce0b88478990f889309L119-R189)
* [`src/components/EventCards/__tests__/EventOptionsDropdown.test.js`](diffhunk://#diff-417d89c0829b57c4cd0c5016d44f8a478675fbb20b420f06e5dc9ffae11d94c4R7): Replaced `userEvent.click` with `user.click` for consistency with updated user-event setup. [[1]](diffhunk://#diff-417d89c0829b57c4cd0c5016d44f8a478675fbb20b420f06e5dc9ffae11d94c4R7) [[2]](diffhunk://#diff-417d89c0829b57c4cd0c5016d44f8a478675fbb20b420f06e5dc9ffae11d94c4R18-R21) [[3]](diffhunk://#diff-417d89c0829b57c4cd0c5016d44f8a478675fbb20b420f06e5dc9ffae11d94c4L49-R54)
* [`src/components/EventCards/__tests__/LoggableEventCard.test.js`](diffhunk://#diff-e2581b4e664fcbb90bcd438da7958521566797e80b3e0dc143cc830eac8c6a99L1-R1): Updated the date adapter to `AdapterDateFns`. [[1]](diffhunk://#diff-e2581b4e664fcbb90bcd438da7958521566797e80b3e0dc143cc830eac8c6a99L1-R1) [[2]](diffhunk://#diff-e2581b4e664fcbb90bcd438da7958521566797e80b3e0dc143cc830eac8c6a99L29-R29)

### CODE REVIEW

File: src/components/EventCards/EventDatepicker.tsx [hunk 1]
```diff
@@ -84,17 +85,15 @@ const EventDatepicker = ({ eventId, isShowing, disabledDates, onAccept }: Props)
                 >
                     <DatePicker
                         label="Event date"
-                        format="MM/D/YYYY"
+                        format="MM/d/yyyy"
```
[the remark this answers is not in the document]
Comment (nathanchica) [reply src/components/EventCards/EventDatepicker.tsx:88]: project is still in development so no real users
